openfl-workspace/torch/histology_s3/plan/datasources.yaml

```diff
@@ -0,0 +1,6 @@
+# Copyright (C) 2025 Intel Corporation
+# Licensed subject to the terms of the separately executed evaluation license agreement between Intel Corporation and you.
+
+bob, plan/datasources_conf.json
+charlie,plan/datasources_2_conf.json
```

Comment:
Instead of creating a separate yaml file for data sources, I'd suggest enhancing the functionality around `data.yaml` (incl. the `fx collaborator create -d` command) - to accommodate for (optional) datasources defined as `json` configurations. This could be done for example by adding a `-ds`/`--data_sources` CLI parameter (see [documentation reference](https://openfl.readthedocs.io/en/latest/reference/fx.collaborator.html#fx-collaborator-create)).

Note that the names of the collaborators should not be present in the `yaml` files of the workspace template, as they are defined when configuring the actual workspace (see the [quickstart guide](https://openfl.readthedocs.io/en/latest/tutorials/taskrunner.html) for reference).

Reply:
@teoparvanov let's talk when you are back.